master/internal/rm/kubernetesrm/request_queue.go

```diff
@@ -99,128 +92,133 @@ type requestQueue struct {
 	podInterfaces       map[string]typedV1.PodInterface
 	configMapInterfaces map[string]typedV1.ConfigMapInterface
 
-	queue                    []*queuedResourceRequest
-	pendingResourceCreations map[*actor.Ref]*queuedResourceRequest
-	availableWorkers         []*actor.Ref
+	mu         sync.Mutex
+	workerChan chan interface{}
+
+	queue []*queuedResourceRequest
+
+	creationInProgress       set.Set[callbackPtr]
+	pendingResourceCreations map[callbackPtr]*queuedResourceRequest
+	blockedResourceDeletions map[callbackPtr]*queuedResourceRequest
+
+	availableWorkers int
+
+	syslog *logrus.Entry
+}
 
-	creationInProgress       set.Set[*actor.Ref]
-	blockedResourceDeletions map[*actor.Ref]*queuedResourceRequest
+func getCallbackPtr(handler errorCallbackFunc) callbackPtr {
+	return (callbackPtr)(reflect.ValueOf(handler))
 }
```

Comment:
i assume if we switch to non-func pointer ids, this goes away. but why did we need this `reflect` usage, anyway?